macros.in

```diff
@@ -1264,31 +1330,48 @@ end}
 # Automatically apply all patches
 # Patches are applied in the order they are listed in the spec file
 # not by their number!
+# -p<num>       Use -p<num> for patch application
 # -m<min>       Apply patches with number >= min only
 # -M<max>       Apply patches with number <= max only
+# -v            Be verbose
 %autopatch(vp:m:M:)\
-%{lua:\
-local options = rpm.expand("%{!-v:-q} %{-p:-p%{-p*}} ")\
-local low_limit = tonumber(rpm.expand("%{-m:%{-m*}}"))\
-local high_limit = tonumber(rpm.expand("%{-M:%{-M*}}"))\
-for i, p in ipairs(patches) do\
-    local inum = patch_nums[i]\
-    if ((not low_limit or inum>=low_limit) and (not high_limit or inum<=high_limit)) \
-    then\
-        print(rpm.expand("%apply_patch -m %{basename:"..p.."}  "..options..p.." "..i.."\\n")) \
-    end\
-end}
+%{expand:\
+  %%{?__scm_apply_patches_%{__scm}:\
+     %%{__scm_apply_patches_%{__scm} %{!-v:-q} \
+                                     %{-p:-p%{-p*}} \
+                                     %{-m:-m%{-m*}} \
+                                     %{-M:-M%{-M*}} \
+     }\
+  }\
+  %%{!?__scm_apply_patches_%{__scm}:\
+     %%{lua:\
+        local options = rpm.expand("%%{!-v:-q} %%{-p:-p%%{-p*}} ")\
+        local low_limit = tonumber(rpm.expand("%%{-m:%%{-m*}}"))\
+        local high_limit = tonumber(rpm.expand("%%{-M:%%{-M*}}"))\
+        for i, p in ipairs(patches) do\
+            local inum = patch_nums[i]\
+            if ((not low_limit or inum>=low_limit) and (not high_limit or inum<=high_limit)) \
+            then\
+                print(rpm.expand("%%apply_patch -m %%{basename:"..p.."}  "..options..p.." "..i.."\\n")) \
+            end\
+        end\
+     }\
+  }\
+}
```

Comment:
This *huge* %{expand:...} with all those escaped %'s is a bit unwieldy. %{expand:...} is better limited to the smallest possible scope because those expansions and escapes add up (see eg #313 for context).

Also makes me wonder if it'd be saner to handle this by making __scm_apply_patches_SCM mandatory for all and just point all the other SCM's to the generic %{lua:} bit (call it __scm_apply_patches_gen or whatever). 

Dunno, but this construct does make me feel uneasy.